scripts/TRN(registry data).R

```diff
@@ -0,0 +1,415 @@
+
+# Columns we want cleaned by Maia's script
+# got rid of WHO number column (who_universal_trial_reference_number_utrn) as Maia said to keep it separate, plus the well formatted numbers here still return NA with which_registry()
+# which will mess up our logic
+
+protocol_columns_to_clean <- c("isrctn_number",
+                               "nct_number",
+                               "other_ids")
+
+# Initialize separate unclean_ids columns for each column to be cleaned
+for (col in protocol_columns_to_clean) {
+  unclean_col <- paste0(col, "_protocol_unclean")
+  EU_protocol_clean[[unclean_col]] <- NA_character_
+}
+
+# Add progress bar (Be patient, progress bar takes at least 10 minutes to display at all)
+
+cli_progress_bar(name = "Cleaning EU protocol data columns (3 total) : ", total = 3)
+
+# Clean table column by column
+for (col in protocol_columns_to_clean) {
+
+
+  cleaned_trns <- vector("list", length <- nrow(EU_protocol_clean))
+
+  # 'trn' is the id in each row currently being cleaned
+  for (i in 1:nrow(EU_protocol_clean)) {
+
+    trn <- EU_protocol_clean[i, col]
+
+    # Detects whether cleaning the string 'trn' throws an error. If yes, initialize 'cleaned' with "Error"
+    cleaned <- tryCatch(clean_trn(trn, quiet = TRUE), error = function(e) "Error")
+
+    # If the TRN can't be cleaned, eliminate it from cleaned column and place in corresponding unclean column for later evaluation
+    if(cleaned == "Error" | is.na(trn)) {
+      cleaned_trns[[i]] <- NA
+      unclean_col <- paste0(col, "_protocol_unclean")
+      EU_protocol_clean[[unclean_col]][i] <- trn # Could be either a garbage number or NA
+    }
+
+    # If 'trn' can be cleaned, save it and discard the old TRN
+    else {
+      cleaned_trns[[i]] <- cleaned
+      unclean_col <- paste0(col, "_protocol_unclean")
+      EU_protocol_clean[[unclean_col]][i] <- NA
+    }
+  }
+  # Cleaned TRNs are placed back in the column they belong
+  EU_protocol_clean[, col] <- unlist(cleaned_trns)
+
+  # Update progress bar
+  cli_progress_update()
+}
+
+# Terminate progress bar
+cli_progress_done()
+
+##########################################################
+## Clean the last few stragglers not caught by algorithm and put in trns_reg
+
+# 2011-000911-26 has 'DKRS00000766' (typo) in other_identifiers, won't be caught by algorithm
+# 2010-019181-91 has valid NCT and DRKS number, prob easier just to enter manually
+# also 2013-000577-77 ; 2020-005450-18
+
+# DRKS stragglers have format 'Name: DRKS Number: 00003246'
+# NCT stragglers have format 'Name: NCT Number: 03351608'
+
+# Function to identify and clean DRKS stragglers
+clean_drks_number <- function(string) {
+  drks_match <- regmatches(string, regexpr("DRKS Number: (\\d+)", string))
+
+  # If a match is found, return the cleaned value, otherwise, return NA
+  if (length(drks_match) < 1) {
+    return(NA)
+  }
+  else if (length(drks_match[[1]]) > 0) {
+    cleaned_value <- paste0("DRKS", gsub("\\D", "", drks_match[[1]]))
```

Comment:
`drks_match <- regmatches(string, regexpr("DRKS Number: (\\d+)", string))`

Since the regmatches() function returns a list of vectors, we need to access the first element of drks_match for the cleaned DRKS number :)